### ISSUE TITLE
Novatel driver in C++ supporting more data frames

### ISSUE THREAD

**Issue by jlblancoc** [ros-drivers/novatel_span_driver#22]
Hi all, 

I recently found this package for interfacing Novatel  devices, and since it seems that it "only" supports these [5](https://github.com/ros-drivers/novatel_span_driver/blob/master/novatel_span_driver/src/novatel_span_driver/mapping.py) Novatel frames, I thought it might be a good idea to let you know about the existence of a more complete C++ implementation we've been developing during the last couple of years at our lab. 

These are the relevant C++ classes that do all the work:
* [mrpt::hwdrivers::CGPSInterface](http://mrpt.ual.es/reference/devel/classmrpt_1_1hwdrivers_1_1_c_g_p_s_interface.html): The interface to the serial port(s). It has parameters to enable the NMEA and Novatel (**binary** frames) parsers. By default, both are automatically detected. It integrates optional ways to send RTK correction streams to the receiver, via an external NTRIP server, for example.

![mrpt_gps_classes_usage](https://user-images.githubusercontent.com/5497818/29146758-53bf105e-7d63-11e7-9980-dd5bdcb88ab1.png)

* [mrpt::obs::CObservationGPS](http://mrpt.ual.es/reference/devel/classmrpt_1_1obs_1_1_c_observation_g_p_s.html): Readings are returned as objects of this type. 

* The list of known NMEA and Novatel OEM6 messages [is here](http://mrpt.ual.es/reference/devel/namespacemrpt_1_1obs_1_1gnss.html#a5f244678709db12be8295a416e4f1459) and the Novatel binary frames which are parsed and stored as C++ structs are defined [here](http://mrpt.ual.es/reference/devel/gnss__messages__novatel_8h_source.html).

I don't have time for it right now, but perhaps someone else with spare time and the motivation would like to experiment and build an alternative ROS node for Novatel receivers based on these classes... ;-) In that case, I'll be glad to help solving doubts regarding our MRPT classes and data structures.

Cheers,
JL

**Comment by JWhitleyAStuff** [comment]
@jlblancoc - I am doing my best to get this driver back to "actively maintained" status. However, friends of mine at Southwest Research Institute just recently released a driver that may be exactly what you're looking for (they're also actively looking for feedback/testing). Please see https://github.com/swri-robotics/novatel_gps_driver and leave some feedback here as well on how it compares to this driver, if you wouldn't mind. Thanks.

**Comment by jlblancoc** [comment]
Hi @JWhitleyAStuff , 

Thanks for the pointer! It seems clear to me that SWRI's is now far more advanced and complete, so in case we need more features, we would probably test their package and propose PR's there.

The only thing I miss in SWRI's package is the possibility of connecting to an NTRIP RTK correction source, directly integrated into the node(lets). But after a quick search, it seems this could be done with such a small script as [this one](https://github.com/tilk/ntrip_ros), so that's not a big deal...

In short: I think I should close this ticket since in the end we would end up duplicating efforts in maintaining two packages for the same thing, right? Also, we'll test at our lab SWRI's package and provide feedback to them.
